### PR TITLE
more algorithm annotations

### PR DESCRIPTION
**Pull request by ktarbet**
## Problem Description

Adding more Annotations to Algorithms

One case I could not drop the  _inputNames[] and String _outputNames[] because that are used in the actual algo.  So I kept the associated get____ methods;

### CODE REVIEW

File: src/main/java/decodes/tsdb/algo/ExpressionParserAlgorithm.java [hunk 1]
```diff
@@ -429,12 +404,8 @@ else if (out1value instanceof String)
 			else if (out2value instanceof String)
 				setOutput(out2, (String)out2value);
 		}
-//AW:AFTER_TIMESLICES_END
 	}
 
-	/**
-	 * Required method returns a list of all input time series names.
-	 */
 	public String[] getInputNames()
```
Comment (MikeNeilson) [inline src/main/java/decodes/tsdb/algo/ExpressionParserAlgorithm.java:409]: This method is defined in the super class, you can remove this and the array and just call this.getInputNames when needed, or cache it during beforeTimeSlices as appropriate.

File: src/main/java/decodes/tsdb/algo/ExpressionParserAlgorithm.java [hunk 2]
```diff
@@ -1,61 +1,47 @@
...
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Comparator;
-import java.util.Date;
-import java.util.Observable;
-import java.util.Observer;
-import java.util.Properties;
-
-import org.nfunk.jep.SymbolTable;
```
Comment (MikeNeilson) [inline src/main/java/decodes/tsdb/algo/ExpressionParserAlgorithm.java:11]: and this still compiles?

Comment (ktarbet) [reply src/main/java/decodes/tsdb/algo/ExpressionParserAlgorithm.java:11]: I used my IDE to cleanup the imports, import org.nfunk.jep.SymbolTable; is still included.

File: src/main/java/decodes/tsdb/algo/ExpressionParserAlgorithm.java (hunk 1 above)
Comment (ktarbet) [reply src/main/java/decodes/tsdb/algo/ExpressionParserAlgorithm.java:409]: What do you think about caching in the super class;  the inputs, and outputs, should not be changing should they?

File: src/main/java/decodes/tsdb/algo/HdbEvaporation.java [hunk 3]
```diff
@@ -1,61 +1,38 @@
...
+		"Primary value is \"area\".\n" +
+		"Really need average area, so two choices for second input:\n" +
+		"prevArea = previous EOP area or diffArea = deltaArea\n" +
+		"For now, chose prevArea.\n" +
+		"Third input is evapCoeff. This is a SDI pointer to either\n" +
+		"a coefficient timeseries, or to lookup a coefficient from the stat tables\n" +
+		"Output is evaporation as a volume.\n" +
+		"Properties include: \n" +
+		"\n" +
+		"ignoreTimeSeries - completely ignore changes to any timeseries value from evapCoeff, and always lookup from database.\n" +
```
Comment (ktarbet) [inline src/main/java/decodes/tsdb/algo/HdbEvaporation.java:19]: I removed the Html markup previously used for Javadoc's, but not inside annotation;  I don't think the text box will render Html?

Comment (ktarbet) [reply src/main/java/decodes/tsdb/algo/HdbEvaporation.java:19]: We could keep javadocs also, but that feels like duplication.

Comment (MikeNeilson) [reply src/main/java/decodes/tsdb/algo/HdbEvaporation.java:19]: We could also make that box render HTML; Swing has some support for the basics. But we'd need some way identify that it is HTML.

**Review by MikeNeilson — APPROVED** [review]

File: src/main/java/decodes/tsdb/algo/HdbEvaporation.java (hunk 3 above)
Comment (ktarbet) [reply src/main/java/decodes/tsdb/algo/HdbEvaporation.java:19]: I'll keep the HTML for future rendering if HTML is detected, I don't think Html will hurt in text rendering.  Many algos only have text in the javadocs

**Comment by ktarbet** [comment]
I investigated adding Annotations to PythonAlgorithm.java, but It seems the Inputs and outputs are defined elsewhere for a using or creating a custom Python Algorithm.

**Review by MikeNeilson — APPROVED** [review]

**Comment by ktarbet** [comment]
A few Algorithms that I've seen seem to have two layers of Properties.  Do we need both?

* the String _propertyNames[]  (that can be replaced with @org.opendcs.annotations.PropertySpec(value="?")
* protected PropertySpec[] getAlgoPropertySpecs() -- that is defined separately but similar to the annotations. but is explicit on the data type for example: PropertySpec.BOOLEAN

Specific Example  :  PeriodToDate.java

```java
private PropertySpec ratingPropertySpecs[] = 
	{
		new PropertySpec("goodQualityOnly", PropertySpec.BOOLEAN,
			"(default=false) Only include good quality values in the calculation.")
	};
	@Override
	protected PropertySpec[] getAlgoPropertySpecs()
	{
		return ratingPropertySpecs;
	}

	@org.opendcs.annotations.PropertySpec(value="false")
	public boolean goodQualityOnly = false;
```

File: src/main/java/decodes/tsdb/algo/ExpressionParserAlgorithm.java (hunk 1 above)
Comment (MikeNeilson) [reply src/main/java/decodes/tsdb/algo/ExpressionParserAlgorithm.java]: We'll have to think about that. so long as it's an instance variable that should work without issue, I think.

**Comment by MikeNeilson** [comment]
> I investigated adding Annotations to PythonAlgorithm.java, but It seems the Inputs and outputs are defined elsewhere for a using or creating a custom Python Algorithm.

Yeah, that's not super trivial. You *can* add Java annotations to Jython code, but we don't structure our Python Algorithms in a way that would work.

**Pull request merged**